warehouse/templates/includes/packaging/project-data.html

```diff
@@ -43,12 +43,23 @@ <h6>{% trans %}Maintainers{% endtrans %}</h6>
       </span>
     {% endfor %}
   {% endif %}
+  {% if release.github_repo_info_url and release.github_open_issue_info_url and release.is_source_verified %}
+    <h6>{% trans %}Project links{% endtrans %}</h6>
+    {% set url = release.urls["Homepage"] %}
+    {% if is_valid_uri(url) %}
+    <li>
+      <a class="vertical-tabs__tab vertical-tabs__tab--with-icon vertical-tabs__tab--condensed" href="{{ url }}" rel="nofollow">
+        {{ url_icon("github.com", "github.com") }}Source <i class="fa fa-circle-check check"></i>
+      </a>
+    </li>
+    {% endif %}
+  {% endif %}
 </div>
 
 <div class="sidebar-section verified">
   <h3 class="sidebar-section__title">{% trans %}Unverified details{% endtrans %}</h3>
   <small><i>These details have <b>not</b> been verified by PyPI</i></small>
-{% if release.urls.values() | contains_valid_uris %}
+{% if release.urls.values() | contains_valid_uris and not release.is_source_verified %}
```

Comment:
This new check should be moved down a few lines (where `is_valid_uri` is being checked) because we only want to skip if if it's a verified 'Source' URL, we don't want to skip the entire section.